### PR TITLE
generate Swagger doc from Godoc

### PR DESCRIPTION
**Pull request by stevekuznetsov**
This PR leverages upstream tools to generate Swagger doc from our Godoc.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@smarterclayton the tool is picking up our `{Type,List}Meta` as a field `'metadata'` - is this an issue?

**Comment by stevekuznetsov** [comment]
@pravisankar @rajatchopra PTAL SDN doc changes

**Comment by smarterclayton** [comment]
Yeah, I think it may be because we don't have godoc on type meta?

**Comment by smarterclayton** [comment]
You don't have to do everything in this pull - let's get the core code and the example output and then break the actual changes into chunks.

**Comment by stevekuznetsov** [comment]
@smarterclayton Should we have Godoc on `TypeMeta`? Upstream doesn't seem to have it, they `inline` it as well. I am going to do all of the `description` tag -> comment moves in this pull, anything that takes more work will be whitelisted so the `hack` script doesn't fail.

**Comment by smarterclayton** [comment]
Hrm, then it's something else.  We don't want to add it if upstream has it.

**Comment by smarterclayton** [comment]
* doesn't have it

**Comment by stevekuznetsov** [comment]
@smarterclayton @liggitt please review while it's hot. this is going to be fun to keep current

**Comment by stevekuznetsov** [comment]
we can drop upstream commit for now, one review is over I will remove it. maybe I'll spend some time upstream making the tool better later

**Comment by smarterclayton** [comment]
[test]

**Comment by stevekuznetsov** [comment]
@smarterclayton removed generated docs

**Comment by smarterclayton** [comment]
Waiting for travis and then out comes the merge hammer.

**Comment by deads2k** [comment]
> Waiting for travis and then out comes the merge hammer.

I'd like a hold on this until after the rebase lands.  I've got an lgtm, waiting to see it go green.

**Comment by stevekuznetsov** [comment]
@smarterclayton pared this down to only the tool addition and script update

File: hack/gen-swagger-docs.sh [hunk 1]
```diff
@@ -7,10 +7,56 @@ set -o nounset
 set -o pipefail
```
Comment (smarterclayton) [inline hack/gen-swagger-docs.sh]: shouldn't this be `hack/update-generated-swagger-docs.sh`?

File: hack/verify-generated-swagger-descriptions.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+echo "===== Verifying Generated Swagger Descriptions ====="
+
+VERIFY=true "${OS_ROOT}/hack/update-generated-swagger-descriptions.sh"
```
Comment (stevekuznetsov) [inline hack/verify-generated-swagger-descriptions.sh]: @deads2k how do you like this for DRYing out our verify scripts?

Comment (smarterclayton) [inline hack/verify-generated-swagger-descriptions.sh]: Inconsistent with upstream worries me, but I'm fine with simpler verifiers.  Duplication has bit us a few times there.

**Comment by pravisankar** [comment]
SDN doc changes LGTM

File: hack/verify-generated-swagger-descriptions.sh (hunk 2 above)
Comment (stevekuznetsov) [inline hack/verify-generated-swagger-descriptions.sh]: If I have the bandwidth I'll push this reduction in boilerplate upstream, too.

**Comment by stevekuznetsov** [comment]
@smarterclayton pending Jenkins tests, anything else this needs?

**Comment by stevekuznetsov** [comment]
[test] me 

@smarterclayton PTAL, will squash if looks ok

**Comment by smarterclayton** [comment]
Your stuff is so good you made explain useful!  Go ahead and fix that.

**Comment by stevekuznetsov** [comment]
@smarterclayton yep, just caught that.

**Comment by stevekuznetsov** [comment]
etcd flake [here](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/11735/consoleFull#-96769135256bdfb7fe4b05b79524e5734):
```
F0224 11:26:30.193373   28241 auth.go:142] error #0: client: etcd member http://127.0.0.1:44001 has no leader
```
 https://github.com/openshift/origin/issues/7243

re[test]

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**